src/plugins/discover/public/components/logs_explorer_tabs/logs_explorer_tabs.tsx

```diff
@@ -29,17 +36,31 @@ export const LogsExplorerTabs = ({ services, selectedTab }: LogsExplorerTabsProp
   const discoverUrl = discoverLocator?.getRedirectUrl(emptyParams);
   const logsExplorerUrl = logsExplorerLocator?.getRedirectUrl(emptyParams);
 
-  const navigateToDiscover = createNavigateHandler(() => {
-    if (selectedTab !== 'discover') {
-      discoverLocator?.navigate(emptyParams);
+  const [_, setLastUsedViewer] = useLocalStorage<
+    typeof DISCOVER_APP_ID | typeof OBSERVABILITY_LOGS_EXPLORER_APP_ID
+  >(OBS_LOGS_EXPLORER_LOGS_VIEWER_KEY, OBSERVABILITY_LOGS_EXPLORER_APP_ID);
+
+  const navigateToDiscover = createNavigateHandler(
+    () => {
+      if (selectedTab !== 'discover') {
+        discoverLocator?.navigate(emptyParams);
+      }
+    },
+    () => {
+      setLastUsedViewer(DISCOVER_APP_ID);
     }
```

Comment:
```suggestion
    () => {
      setLastUsedViewer(DISCOVER_APP_ID);
      if (selectedTab !== 'discover') {
        discoverLocator?.navigate(emptyParams);
      }
    }
```
Nit: why not just make these calls directly in the `onClick` callback vs a dedicated `updateState` callback?

Reply:
Thanks for the suggestion. I've incorporated the change.